lib/BattleHex.cpp

```diff
@@ -102,58 +102,54 @@ BattleHex& BattleHex::moveInDir(EDir dir, bool hasToBeValid)
 	default:
 		throw std::runtime_error("Disaster: wrong direction in BattleHex::operator+=!\n");
 		break;
-}
+	}
 	return *this;
 }
 
 BattleHex &BattleHex::operator+=(BattleHex::EDir dir)
 {
-	return moveInDir(dir);
+	return moveInDirection(dir);
 }
 
-BattleHex BattleHex::movedInDir(BattleHex::EDir dir, bool hasToBeValid) const
+BattleHex BattleHex::cloneInDirection(BattleHex::EDir dir, bool hasToBeValid) const
 {
-	BattleHex result(*this);
-	result.moveInDir(dir, hasToBeValid);
+	BattleHex result(hex);
+	result.moveInDirection(dir, hasToBeValid);
 	return result;
 }
 
 BattleHex BattleHex::operator+(BattleHex::EDir dir) const
 {
-	return movedInDir(dir);
+	return cloneInDirection(dir);
 }
 
 std::vector<BattleHex> BattleHex::neighbouringTiles() const
 {
 	std::vector<BattleHex> ret;
-	const int WN = GameConstants::BFIELD_WIDTH;
-	// H3 order : TR, R, BR, BL, L, TL (T = top, B = bottom ...)
-
-	checkAndPush(hex - ( (hex/WN)%2 ? WN+1 : WN ), ret); // 1
-	checkAndPush(hex + 1, ret); // 2
-	checkAndPush(hex + ( (hex/WN)%2 ? WN : WN+1 ), ret); // 3
-	checkAndPush(hex + ( (hex/WN)%2 ? WN-1 : WN ), ret); // 4
-	checkAndPush(hex - 1, ret); // 5
-	checkAndPush(hex - ( (hex/WN)%2 ? WN : WN-1 ), ret); // 6
-
+	checkAndPush(cloneInDirection(TOP_LEFT, true), ret);
+	checkAndPush(cloneInDirection(TOP_RIGHT, true), ret);
+	checkAndPush(cloneInDirection(RIGHT, true), ret);
+	checkAndPush(cloneInDirection(BOTTOM_RIGHT, true), ret);
+	checkAndPush(cloneInDirection(BOTTOM_LEFT, true), ret);
+	checkAndPush(cloneInDirection(LEFT, true), ret);
 	return ret;
 }
 
 signed char BattleHex::mutualPosition(BattleHex hex1, BattleHex hex2)
 {
-	if(hex2 == hex1 - ( (hex1/17)%2 ? 18 : 17 )) //top left
-		return 0;
-	if(hex2 == hex1 - ( (hex1/17)%2 ? 17 : 16 )) //top right
-		return 1;
-	if(hex2 == hex1 + 1 && hex1%17 != 16) //right
-		return 2;
-	if(hex2 == hex1 + ( (hex1/17)%2 ? 17 : 18 )) //bottom right
-		return 3;
-	if(hex2 == hex1 + ( (hex1/17)%2 ? 16 : 17 )) //bottom left
-		return 4;
-	if(hex2 == hex1 - 1 && hex1%17 != 0) //left
-		return 5;
-	return -1;
+	if(hex2 == hex1.cloneInDirection(TOP_LEFT,true))

```

Comment:
It's good look to the header file, TOP_LEFT->0, TOP_RIGHT->1, RIGHT->2.... I check everything and it's work. You want use enum so I use it.